### ISSUE TITLE
Timeout error - large number of search results

### ISSUE THREAD

**Issue by shivaramakrishna99** [MaayanLab/DRC-Portals#386]
The below error message needs to be replaced with something more human readable:

<img width="1177" alt="image" src="https://github.com/user-attachments/assets/6b93d805-c0df-443e-a2e8-583942f1f86c">

**Comment by u8sand** [comment]
Agreed, this comes from: https://github.com/MaayanLab/DRC-Portals/commit/3d0998308db63270600e518464db076219ff5806#diff-4b206a60e9d28b9a742ac512a5772d4ec5eebb1e0085deed9e833a3067b69518R275

Typically we can log the error to the console but display something else to the user, like: "An error occurred while executing this query, please try again later." or if we handle this specific timeout we can say "Please try to narrow down your search"

**Comment by susrinivasan** [comment]
Hi Daniel, We added this to log the error to make it more meaningful. However, we could not test this snippet since we are unable to recreate this error on our branch. Since both our locahost and the production portal are pointing to the same DB server, we assume that this error is due to node.js/npm server timeout. On our instance, it takes about 50 secs to 1 minute to populate the table when we search for cancer and if the time out is less than that, it probably throws the error. We request you to pull in our changes and we can see whether our new message appears. THanks.
catch (error) {
        console.error("Error fetching C2M2 Main table:", error);
    
        // Check if the error is a timeout error
        if (error instanceof Error && error.message.includes("Timed out")) {
            console.error("Error fetching C2M2 Main table:", error);
            return <div>This error was caused by a timeout. Please narrow down your search.</div>;
        }
    
        // General error handling
        return <div>Error fetching C2M2 Main table: {(error as Error).message}</div>;
    }
